libs/metrics/in_out.py

```diff
@@ -1,61 +1,263 @@
+            total_in += path_in
+            total_out += path_out
+
+        # Normalize in_out:
+        if total_in > total_out:
+            total_in, total_out = 1, 0
+        elif total_in < total_out:
+            total_in, total_out = 0, 1
+        elif total_in > 1: # total_in == total_out
+            total_in, total_out = 1, 1
+        # else total_in == total_out = (0,0) or (1,1), number is already normalized.
+        return total_in, total_out
 
     @classmethod
     def generate_daily_csv_data(cls, yesterday_hourly_file):
-        raise NotImplementedError("Operation not implemented")
+        people_in, people_out = 0, 0
+        with open(yesterday_hourly_file, newline='') as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                people_in += int(row["In"])
+                people_out += int(row["Out"])
+        return people_in, people_out
 
     @classmethod
-    def generate_live_csv_data(cls, today_entity_csv, entity, entries_in_interval):
+    def generate_hourly_metric_data(cls, config, objects_logs, entity):
+        boundary_line = cls.retrieve_in_out_boundary(config, entity["id"])
+        summary = np.zeros((len(objects_logs), 2), dtype=np.long)
+        for index, hour in enumerate(sorted(objects_logs)):
+            hour_objects_detections = objects_logs[hour]
+            for track_id, data in hour_objects_detections.items():
+                path = data["path"]
+                new_in, new_out = cls.process_path(boundary_line, path)
+                summary[index] += (new_in, new_out)
+        return summary
+
+    @classmethod
+    def generate_live_csv_data(cls, config, today_entity_csv, entity, entries_in_interval):
         """
         Generates the live report using the `today_entity_csv` file received.
         """
-        raise NotImplementedError("Operation not implemented")
+        people_in, people_out = 0, 0
+        boundary_line = cls.retrieve_in_out_boundary(config, entity["id"])
+        with open(today_entity_csv, "r") as log:
+            objects_logs = {}
+            lastest_entries = deque(csv.DictReader(log), entries_in_interval)
+            for entry in lastest_entries:
+                cls.process_csv_row(entry, objects_logs)
+        paths = {}
+        for hour in objects_logs:
+            for track_id, sub_path in objects_logs[hour].items():
+                if track_id not in paths:
+                    paths[track_id] = sub_path
+                else:
+                    paths[track_id].extend(sub_path)
+        for track_id, data in paths.items():
+            path = data["path"]
+            new_in, new_out = cls.process_path(boundary_line, path)
+            people_in += new_in
+            people_out += new_out
+        return [people_in, people_out]
 
     @classmethod
     def get_trend_live_values(cls, live_report_paths: Iterator[str]) -> Iterator[int]:
```

Comment:
Are we showing the indicator trend in the dashboard? I am not sure if makes sense to calculate it if is not used in any place.

Reply:
I don't think so, I don't think it hurts too much do a "basic version" just to simplify its integration